include/lsst/afw/image/TransmissionCurve.h

```diff
@@ -0,0 +1,300 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST/AURA.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_IMAGE_TransmissionCurve_h_INCLUDED
+#define LSST_AFW_IMAGE_TransmissionCurve_h_INCLUDED
+
+#include "ndarray_fwd.h"
+
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/afw/geom/Box.h"
+#include "lsst/afw/table/io/Persistable.h"
+
+namespace lsst {
+namespace afw {
+namespace image {
+
+
+/**
+ *  A spatially-varying transmission curve as a function of wavelength.
+ *
+ *  TransmissionCurve can only be evaluated at discrete user-provided positions;
+ *  it does not provide an interface for computing average transmission over
+ *  regions or computing spatially-varying scalars from integrals over the
+ *  wavelength dimension.
+ *
+ *  TransmissionCurves are immutable and are expected to be passed and held
+ *  by shared_ptr.  As such they are neither copyable nor movable.
```

Comment:
If a class is both immutable and always held by `shared_ptr`, there's no need to copy: you never have to protect yourself from changes made by another holder of the shared pointer, and you never have do protect other holders from changes you might make.  Disabling copying encodes that philosophy at the language level, prevents inadvertent copies of large objects, and (for copy assignment) prevents polymorphic slicing.

Reply:
Unless you're using a framework (e.g., standard containers) that assumes objects are copyable, in which case you need to jump through extra hoops.

Note that I'm not questioning making the class unassignable -- to do otherwise would directly violate immutability.

Reply:
I don't understand the concern with frameworks that expect copyability, especially standard containers; isn't it unquestionably better to use a container of `shared_ptr` for polymorphic objects?  In any case, doing copyability safely on a polymorphic hierarchy actually requires adding a `virtual` clone method (certainly in this hierarchy, where the derived class are at least mostly hidden and only the ABC is accessible), and that would really feel like a confusingly unnecessary addition.